File: nuget/src/ReactUbiquitous.NuGet/Models/Sections/GridSectionConfig.cs
using System.Text.Json.Serialization;

namespace ReactUbiquitous.NuGet.Models.Sections;

/// <summary>Configuration for a CSS grid layout section.</summary>
public class GridSectionConfig : BaseSectionConfig
{
    /// <summary>Initializes a new instance of <see cref="GridSectionConfig"/> and sets the layout discriminator.</summary>
    public GridSectionConfig() { Layout = "grid"; }

    /// <summary>Gets or sets the grid-template-columns CSS property.</summary>
    [JsonPropertyName("gridTemplateColumns")]
    public string? GridTemplateColumns { get; set; }

    /// <summary>Gets or sets the grid-template-rows CSS property.</summary>
    [JsonPropertyName("gridTemplateRows")]
    public string? GridTemplateRows { get; set; }

    /// <summary>Gets or sets the gap between grid cells.</summary>
    [JsonPropertyName("gap")]
    public string? Gap { get; set; }

    /// <summary>Gets or sets the row-gap between grid rows.</summary>
    [JsonPropertyName("rowGap")]
    public string? RowGap { get; set; }

    /// <summary>Gets or sets the column-gap between grid columns.</summary>
    [JsonPropertyName("columnGap")]
    public string? ColumnGap { get; set; }

    /// <summary>Gets or sets the align-items CSS property for the grid.</summary>
    [JsonPropertyName("alignItems")]
    public string? AlignItems { get; set; }

    /// <summary>Gets or sets the justify-items CSS property for the grid.</summary>
    [JsonPropertyName("justifyItems")]
    public string? JustifyItems { get; set; }
}

File: src/DTOs/Sections/GridSectionConfig.php
<?php

namespace ReactUbiquitous\DTOs\Sections;

final class GridSectionConfig extends BaseSectionConfig
{
    public function __construct(
        string $id,
        array $elements = [],
        ?string $title = null,
        ?string $description = null,
        ?int $order = null,
        ?string $className = null,
        ?array $style = null,
        public readonly ?string $gridTemplateColumns = null,
        public readonly ?string $gridTemplateRows = null,
        public readonly string|int|null $gap = null,
        public readonly string|int|null $rowGap = null,
        public readonly string|int|null $columnGap = null,
        public readonly ?string $alignItems = null,
        public readonly ?string $justifyItems = null,
    ) {
        parent::__construct($id, $elements, $title, $description, $order, $className, $style);
    }

    public function getLayout(): string { return 'grid'; }

    public function toArray(): array
    {
        return array_merge($this->baseArray(), array_filter([
            'gridTemplateColumns' => $this->gridTemplateColumns,
            'gridTemplateRows' => $this->gridTemplateRows,
            'gap' => $this->gap,
            'rowGap' => $this->rowGap,
            'columnGap' => $this->columnGap,
            'alignItems' => $this->alignItems,
            'justifyItems' => $this->justifyItems,
        ], fn($v) => $v !== null));
    }
}
